chromadb/config.py

```diff
@@ -225,11 +268,141 @@ def stop(self) -> None:
     def reset_state(self) -> None:
         """Reset the state of this system and all constituents in reverse dependency order"""
         if not self.settings.allow_reset:
-            raise ValueError("Resetting is not allowed by this configuration (to enable it, set `allow_reset` to `True` in your Settings() or include `ALLOW_RESET=TRUE` in your environment variables)")
+            raise ValueError(
+                "Resetting is not allowed by this configuration (to enable it, set `allow_reset` to `True` in your Settings() or include `ALLOW_RESET=TRUE` in your environment variables)"
+            )
         for component in reversed(list(self.components())):
             component.reset_state()
 
 
+class BasicAuthClientProvider(ClientAuthProvider):
+    _basic_auth_token: SecretStr
+
+    def __init__(self, settings: "Settings") -> None:
+        super().__init__(settings)
+        self._settings = settings
+        if os.environ.get("CHROMA_CLIENT_AUTH_BASIC_USERNAME") and os.environ.get(
+            "CHROMA_CLIENT_AUTH_BASIC_PASSWORD"
+        ):
+            self._basic_auth_token = _create_token(
+                os.environ.get("CHROMA_CLIENT_AUTH_BASIC_USERNAME", ""),
+                os.environ.get("CHROMA_CLIENT_AUTH_BASIC_PASSWORD", ""),
+            )
+        elif isinstance(
+            self._settings.chroma_client_auth_provider_config, str
```

Comment:
Nit: Comment the cases with 
```
# User supplied a path
# User supplied env vars
# User supplied a config
```

Reply:
@HammadB Actually, I am refactoring this whole mess. It won't scale/adapt to new ways of providing configuration and credentials. Instead, now we'll have two new abstractions (driven by settings with certain, hard-coded for now, precedence - env, file, ... anything else).

Reply:
ah, yeah a "loader" interface makes sense.